frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/toolbar/DiscoverAction.java

```diff
@@ -113,7 +113,10 @@ public void run(ProgressIndicator indicator) {
             .onFailure(e -> VcsNotifier.getInstance(project).notifyError(
                 /* title */ getString("action.GitMachete.DiscoverAction.notification.title.write-file-error"),
                 /* message */ e.getMessage() != null ? e.getMessage() : ""))
-            .onSuccess(__ -> baseEnhancedGraphTable.queueRepositoryUpdateAndModelRefresh());
+            .onSuccess(__ -> {
```

Comment:
Nit: hmm can this be moved into `Task`'s own `onFailure`/`onSuccess`? Then probably `GuiUtils.invokeLaterIfNeeded` in `openMacheteFile` won't be even needed since both those `Task`'s callbacks are executed on UI thread; `openMacheteFile` can be `@UIEffect` and `postWriteRunnable` can be `@UI`

Reply:
answered below